src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -680,12 +680,15 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
                         return true;
                     }
                 }
+            }
 
-                // If ignoring dynamic, compare underlying tuple types
+            if (comparison.HasFlag(TypeCompareKind.IgnoreTupleNames))
+            {
+                // If ignoring tuple element names, compare underlying tuple types
                 if (t2.IsTupleType)
                 {
                     t2 = t2.TupleUnderlyingType;
-                    if (this.Equals(t2, ignoreCustomModifiersAndArraySizesAndLowerBounds, ignoreDynamic)) return true;
+                    if (this.Equals(t2, comparison)) return true;
```

Comment:
The case where `this` is a `TupleTypeSymbol` is handled by `internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison)` in `TupleTypeSymbol`.
It goes into the underlying type of `this`.